File: src/1684. Count the Number of Consistent Strings/main.go
// Solved by @kitanoyoru

package main

import "strings"

func countConsistentStrings(allowed string, words []string) int {
  ans := 0

  for _, word := range words {
    for _, ch := range word {
      res := strings.ContainsRune(allowed, ch)
      if !res {
        ans--
        break
      }
    }
    ans++
  }

  return ans
}

File: src/1684. Count the Number of Consistent Strings/main.c
// SOlved by @kitanoyoru

int countConsistentStrings(char * allowed, char ** words, int wordsSize){
  int s = 0, ans = 0;
  for (int i = 0; allowed[i]; i++) {
    s |= 1 << (allowed[i] - 'a');
  }

  for (int i = 0; i < wordsSize; i++) {
    int c = 0;
    for (int j = 0; words[i][j]; j++) {
      c |= 1 << (words[i][j] - 'a');
    }
    if (s == (c | s)) {
      ans++;
    }
  }

  return ans;
}
